src/modules/DetailsPage/Components/DetailsPage/DetailsPage.tsx

```diff
@@ -1,13 +1,258 @@
+import { Box, Grid2, Stack, ThemeProvider, Typography } from "@mui/material";
 import { useParams } from "react-router-dom";
+import { theme } from "../../../../helperStyles/helperStyles";
+import {
+  BathRoomIcon,
+  BedRoomIcon,
+  DiningRoomIcon,
+  LivingRoomIcon,
+  RefigratorIcon,
+  TvIcon,
+  UnitsIcon,
+  WifiIcon,
+} from "../../../Shared/Components/SvgIcons/SvgIcons";
+import { useContext, useEffect, useState } from "react";
+import { axiosInstance, USER_ROOMS_URLS } from "../../../../services/urls";
+import { GetRoomResponse, Room } from "../../../../interfaces/RoomsInterfaces";
+import { toast } from "react-toastify";
+import roomImg1 from "../../../../assets/images/room-img1.png";
+import roomImg2 from "../../../../assets/images/room-img2.png";
+import roomImg3 from "../../../../assets/images/room-img3.png";
+import BookingCard from "../../../Users-Portal/Component/UsersShared/BookingCard/BookingCard";
+import UserPageTitle from "../../../Users-Portal/Component/UsersShared/UserPageTitle/UserPageTitle";
+import CommentForm from "../../../Shared/Components/CommentForm/CommentForm";
+import { styled } from "@mui/system";
+import ReviewForm from "../../../Users-Portal/Component/ReviewForm/ReviewForm";
+import { AuthContext } from "../../../../Context/Context";
 
-
+const StyledBox = styled(Box)(() => ({
+  display: "flex",
+  width: "90%",
+  marginInline: "auto",
+  border: "1px solid #E5E5E5",
+  borderRadius: "15px",
+  marginBottom: "2rem",
+}));
+const VerticalLine = styled(Box)(({ theme }) => ({
+  width: "1px",
+  backgroundColor: "#203FC7",
+  margin: "0 10px",
+  [theme.breakpoints.down("md")]: {
+    display: "none",
+  },
+  [theme.breakpoints.up("md")]: {
+    display: "block",
+  },
+}));
 export default function DetailsPage() {
-	let params=useParams()
-	console.log(params.roomId)
-	return (
-		<>
-			<div>DetailsPage</div>
-
-		</>
-	);
+  const { loginData } = useContext(AuthContext);
+  let { roomId } = useParams<{ roomId: string }>();
+  const [room, setRoom] = useState<Room>();
+  const facilitiesData = [
+    { icon: <BedRoomIcon />, number: 5, name: "bed room" },
+    { icon: <LivingRoomIcon />, number: 1, name: "living room" },
+    { icon: <BathRoomIcon />, number: 3, name: "bath room" },
+    { icon: <DiningRoomIcon />, number: 1, name: "dining room" },
+    { icon: <WifiIcon />, number: 10, name: "mbp/s" },
+    { icon: <UnitsIcon />, number: 7, name: "unit ready" },
+    { icon: <RefigratorIcon />, number: 2, name: "refrigrator" },
+    { icon: <TvIcon />, number: 4, name: "television" },
+  ];
+
+  useEffect(() => {
+    const getRoom = async () => {
```

Comment:
We will need to handle the loading state and provide a feedback to the user. We can for example make a loading skeleton